qcodes/dataset/sqlite_base.py

```diff
@@ -382,8 +410,109 @@ def perform_db_upgrade_1_to_2(conn: SomeConnection) -> None:
     else:
         raise RuntimeError(f"found {n_run_tables} runs tables expected 1")
 
-    log.info('Succesfully upgraded database version 1 -> 2.')
-    set_user_version(conn, 2)
+
+@upgrader
+def perform_db_upgrade_2_to_3(conn: SomeConnection) -> None:
+    """
+    Perform the upgrade from version 2 to version 3
+    """
+
+    no_of_runs_query = "SELECT max(run_id) FROM runs"
+    no_of_runs = one(atomic_transaction(conn, no_of_runs_query), 'max(run_id)')
+    no_of_runs = no_of_runs or 0
+
+    # Insert a new column, "run_description", and fill it out with information
+    # retrieved from the layouts and dependencies tables
+    # If one run fails, we want the whole upgrade to role back, hence the
+    # entire upgrade is one atomic transaction
+
+    with atomic(conn) as conn:
+        sql = "ALTER TABLE runs ADD COLUMN run_description TEXT"
+        transaction(conn, sql)
+
+        for run_id in range(1, no_of_runs+1):
+
+            paramspecs: Dict[int, ParamSpec] = {}
+
+            rst_query = ("SELECT result_table_name FROM runs "
+                         f"WHERE run_id={run_id}")
```

Comment:
Should this and below be done with ? replacement?

Reply:
Yes. Fixed.